src/utils/arfs_builders/arfs_folder_builders.ts

```diff
@@ -4,13 +4,25 @@ import { ArFSPrivateFolder, ArFSPublicFolder } from '../../arfs_entities';
 import { ArweaveAddress } from '../../types/arweave_address';
 import { CipherIV, DriveKey, FolderID } from '../../types';
 import { ArFSFileOrFolderBuilder } from './arfs_builders';
+import { EID, EntityID } from '../../types/entity_id';
+import { stubEntityID } from '../stubs';
+
+export const ROOT_FOLDER_ID_PLACEHOLDER = 'root folder';
+
+// A utility type to assist with fail-safe decryption of private entities
+export class RootFolderID extends EntityID {
+	constructor() {
+		super(`${stubEntityID}`); // Unused after next line
+		this.entityId = ROOT_FOLDER_ID_PLACEHOLDER;
```

Comment:
Beware all root folder ids will equal for any drive

Reply:
We never query against this. It's just for display purposes. But yes it's sketchy and I don't love it.